pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStoreProvider.java

```diff
@@ -51,7 +52,8 @@
         PersistentTopic originPersistentTopic = (PersistentTopic) subscription.getTopic();
         String pendingAckTopicName = MLPendingAckStore
                 .getTransactionPendingAckStoreSuffix(originPersistentTopic.getName(), subscription.getName());
-
+        ManagedLedgerConfig managedLedgerConfig = originPersistentTopic.getManagedLedger().getConfig();
+        managedLedgerConfig.setCreateIfMissing(true);
```

Comment:
this is not good, we cannot alter the configuration of another component this way.

if "createIfMissing" is required we have to set it upstream

Reply:
Thank you for pointing out.
 I made some changes, please take a look again